lib/redlock/client.rb

```diff
@@ -138,11 +146,18 @@ def unlock(resource, val)
         # Nothing to do, unlocking is just a best-effort attempt.
       end
 
+      def locked?(resource)
+        recover_from_script_flush do
+          @redis.evalsha @check_script_sha, keys: [resource]
```

Comment:
There's no need for a script here, simply call [`Redis.exists()`](https://www.rubydoc.info/gems/redis/Redis:exists):

```suggestion
          @redis.exists? resource
```

and then you can also get rid of the `recover_from_script_flush`